### PR TITLE
Queue framework boilerplate

### PR DESCRIPTION
**Pull request by camertron**
The beginnings of the queue framework, wahoo!

The txgh queue is a set of classes for processing push and delete payloads from Github. Actual queue implementations can be added by way of txgh queue's backend interface, although at the moment there are no backends. In this PR:
1. Backend registration functionality.
2. Loading configuration from a .yml file.
3. Job processing logic.
4. Job statuses (`ok`, `fail`, `retry_with_delay`, `retry_without_delay`).
5. The `Result` class, which wraps a `TxghServer::Response` or a `StandardError`.

Coming up next:
1. Error handling
2. SQS backend

@lumoslabs/platform

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,7 +24,13 @@ matrix:
       <<: *txgh
     - rvm: 2.1.6
       <<: *server
+    - rvm: 2.1.6
```
Comment (jdoconnor) [inline .travis.yml]: 2.1.9 or greater, please

File: queue/lib/txgh-queue.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module TxghQueue
+  autoload :Backends,      'txgh-queue/backends'
```
Comment (jdoconnor) [inline queue/lib/txgh-queue.rb:2]: now i remember the other thing activesupport would have helped with ... autoloading by convention

File: queue/lib/txgh-queue/backends.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+module TxghQueue
+  module Backends
+    autoload :Sqs, 'txgh-queue/backends/sqs'
```
Comment (jdoconnor) [inline queue/lib/txgh-queue/backends.rb:3]: weird place to have an autoload, considering it isn't referenced here

**Review by jdoconnor — COMMENTED** [review]
seems ok. I have a few small tweaks

File: queue/lib/txgh-queue.rb (hunk 2 above)
Comment (camertron) [reply queue/lib/txgh-queue.rb:2]: Yeah, it's just such a _heavy_ dependency...

File: .travis.yml (hunk 1 above)
Comment (camertron) [reply .travis.yml]: Will do.

File: queue/lib/txgh-queue/backends.rb (hunk 3 above)
Comment (camertron) [reply queue/lib/txgh-queue/backends.rb:3]: This was a mistake, should have been in an upcoming PR with all the SQS logic.

**Pull request merged**